### ISSUE TITLE
c++: internal compiler error: Killed on compiling caffe

### ISSUE THREAD

**Issue by Autopawn** [yjxiong/temporal-segment-networks#173]
Executing `bash build_all.sh`, while compiling caffe, script was stopped throwing several of the following errors:

    c++: internal compiler error: Killed (program cc1plus)
    Please submit a full bug report,
    with preprocessed source if appropriate.
    See <file:///usr/share/doc/gcc-5/README.Bugs> for instructions.
    src/caffe/CMakeFiles/caffe.dir/build.make:69980: recipe for target 'src/caffe/CMakeFiles/caffe.dir  /layers/image_data_layer.cpp.o' failed
    make[2]: *** [src/caffe/CMakeFiles/caffe.dir/layers/image_data_layer.cpp.o] Error 4
    make[2]: *** Waiting for unfinished jobs....

After searching this was due to g++ using all available memory and swap (8Gb+8Gb on the machine where it was tested) due to several open jobs, this was checked using `htop`.

This error was solved with the following change:
```bash
#if make -j32 install ; then
if make -j2 install ; then
    echo "Caffe Built."
    echo "All tools built. Happy experimenting!"
    cd ../../../
```

**Comment by Autopawn** [comment]
Closed as duplicate of [#109](https://github.com/yjxiong/temporal-segment-networks/issues/109)

**Comment by yjxiong** [comment]
This is possibly due to OOM while compiling. It can be avoided by lowering the parallelization as you did.
